src/Core/Compiler/CompilerService.cs

```diff
@@ -67,33 +70,61 @@ private QsCompilation UpdateCompilation(ImmutableDictionary<Uri, string> sources
             return loaded.CompilationOutput;
         }
 
+        /// <inheritdoc/>
+        public AssemblyInfo BuildEntryPoint(OperationInfo operation, CompilerMetadata metadatas, QSharpLogger logger, string dllName, string executionTarget = null)
+        {
+            var signature = operation.Header.PrintSignature();
+            var argumentTuple = SyntaxTreeToQsharp.ArgumentTuple(operation.Header.ArgumentTuple, type => type.ToString(), symbolsOnly: true);
+
+            var entryPointUri = new Uri(Path.GetFullPath(Path.Combine("/", $"entrypoint.qs")));
+            var entryPointSnippet = @$"namespace ENTRYPOINT
+                {{
+                    open {operation.Header.QualifiedName.Namespace.Value};
+                    @{BuiltIn.EntryPoint.FullName}()
+                    operation {signature}
+                    {{
+                        return {operation.Header.QualifiedName}{argumentTuple};
+                    }}
+                }}";
```

Comment:
I had some discussions about this with @bettinaheim and @anpaz-msft. The reason I ended up doing it this way is that we want to ignore any `@EntryPoint()` attributes that happen to appear inside snippets (code from notebook cells), and we are doing this by compiling the snippets as a library and suppressing the resulting warning.

If we want to attach the `@EntryPoint()` attribute directly to the operation in the snippet code and compile the snippets as an executable, then we'd have to also go remove the attribute from any other snippet operations where it might exist. I think this is certainly possible by manipulating the syntax tree, as you suggest, and maybe we can change to go that route in the future if needed, but this seemed like the simplest solution at the moment.

One potential benefit of this approach is that in the future it may be possible to reference other Q# projects from IQ#, including Q# library projects. This approach would also give a way to submit those operations to Azure Quantum without having to manually create a wrapper operation in the notebook.

Reply:
Seems reasonable, I agree that this is probably simpler. Thanks for explaining!

> One potential benefit of this approach is that in the future it may be possible to reference other Q# projects from IQ#, including Q# library projects. This approach would also give a way to submit those operations to Azure Quantum without having to manually create a wrapper operation in the notebook.

I haven't tested it, but you could probably still attach the entry point attribute to callables in a reference. I think they all get added to the same syntax tree.